apifest-oauth20/src/main/java/com/apifest/oauth20/persistence/cassandra/CassandraDBManager.java

```diff
@@ -0,0 +1,510 @@
+        session.execute(ACCESS_TOKEN_TABLE_CQL);
+
+        // create sec. indexes (if not exists)
+        session.execute("CREATE INDEX IF NOT EXISTS redirect_uri_idx ON "+KEYSPACE_NAME+"."+AUTH_CODE_TABLE_NAME+" (redirect_uri);");
+    }
+
+    @Override
+    protected void finalize() throws Throwable {
+        CassandraConnector.close();
+    }
+
+    @Override
+    public void storeAccessToken(AccessToken accessToken) {
+        try {
+            // TODO: put tokenExpiration as TTL
+            Insert stmt = QueryBuilder.insertInto(KEYSPACE_NAME, ACCESS_TOKEN_TABLE_NAME)
+                .value("access_token", accessToken.getToken())
+                .value("refresh_token", accessToken.getRefreshToken())
+                .value("expires_in", accessToken.getExpiresIn())
+                .value("scope", accessToken.getScope())
+                .value("type", accessToken.getType())
+                .value("valid", accessToken.isValid())
+                .value("client_id", accessToken.getClientId())
+                .value("code_id", accessToken.getCodeId())
+                .value("user_id", accessToken.getUserId())
+                .value("details", accessToken.getDetails())
+                .value("created", accessToken.getCreated())
+                .value("refresh_expires_in", accessToken.getRefreshExpiresIn())
+            ;
+            // TTL
+            Long tokenExpiration = (accessToken.getRefreshExpiresIn() != null && !accessToken.getRefreshExpiresIn().isEmpty()) ? Long.valueOf(accessToken.getRefreshExpiresIn()) : Long.valueOf(accessToken.getExpiresIn());
+            int ttl = tokenExpiration.intValue();
+            stmt.using(QueryBuilder.ttl(ttl));
+
+            session.execute(stmt);
+        } catch(Throwable e) {
+            log.error(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public void updateAccessTokenValidStatus(String accessToken, boolean valid) {
+        try {
+            Update.Where stmt = QueryBuilder.update(KEYSPACE_NAME, ACCESS_TOKEN_TABLE_NAME)
+                    .with(QueryBuilder.set("valid", valid))
+                    .where(QueryBuilder.eq("access_token", accessToken))
+                    ;
+            session.execute(stmt);
+        } catch(Throwable e) {
+            log.error(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public AccessToken findAccessToken(String accessToken) {
+        try {
+            Select.Where stmt = QueryBuilder.select().from(KEYSPACE_NAME, ACCESS_TOKEN_TABLE_NAME)
+                    .where(QueryBuilder.eq("access_token", accessToken));
+            ResultSet rs = session.execute(stmt);
+            Iterator<Row> iter = rs.iterator();
+            if(iter.hasNext()) {
+                AccessToken atoken = new AccessToken();
+                Row row = iter.next();
+                mapRowToAccessToken(row, atoken);
+                return atoken;
+            }
+        } catch(Throwable e) {
```

Comment:
I've removed all catch, because is thrown no exception by cassandra java driver.
In some update methods, there is a boolean return, I'll return always true, because if there are problems during persistence, cassandra java drive will throw an exception.
Tell  me if this approach is correct for you.